sdk/all/src/main/java/io/opentelemetry/sdk/OpenTelemetrySdk.java

```diff
@@ -114,39 +85,41 @@ public Clock getClock() {
 
   /** Returns the {@link TracerSdkManagement} for this {@link OpenTelemetrySdk}. */
   public TracerSdkManagement getTracerManagement() {
-    return (TracerSdkProvider) ((ObfuscatedTracerProvider) tracerProvider).unobfuscate();
+    return (TracerSdkProvider) ((ObfuscatedTracerProvider) getTracerProvider()).unobfuscate();
   }
 
   /** Returns a new {@link Builder} initialized with the values of this {@link OpenTelemetrySdk}. */
   @Override
   public Builder toBuilder() {
-    return builder()
-        .setTracerProvider(tracerProvider)
-        .setMeterProvider(meterProvider)
+    return new Builder()
+        .setTracerProvider(getTracerProvider())
+        .setMeterProvider(getMeterProvider())
         .setPropagators(getPropagators())
         .setClock(clock)
         .setResource(resource);
   }
 
   /** A builder for configuring an {@link OpenTelemetrySdk}. */
-  public static class Builder implements OpenTelemetryBuilder<Builder> {
+  public static class Builder extends DefaultOpenTelemetry.Builder {
     private Clock clock = MillisClock.getInstance();
```

Comment:
For the below `null` checks to work I think these need to default to `null`

Reply:
very good point! will fix.